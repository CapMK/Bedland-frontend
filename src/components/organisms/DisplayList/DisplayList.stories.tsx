import type { Meta, StoryObj } from '@storybook/react';
import { PARAM_REDUX_MERGE_STATE } from 'addon-redux';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import mockResidentFlats from 'utils/mock/mockResidentsFlats';
import mockBuildingsList from 'utils/mock/mockBuildingsList';
import DisplayList from './DisplayList';

const meta = {
  title: 'Organisms/DisplayList',
  component: DisplayList,
  parameters: {
    backgrounds: { default: 'light' },
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    displayList: {
      table: {
        disable: true,
      },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof DisplayList>;

export default meta;

type Story = StoryObj<typeof meta>;

export const FlatsList: Story = {
  args: { displayList: mockResidentFlats },
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "resident", "loggedIn": "true"}}',
  },
};

export const BuildingsList: Story = {
  args: { displayList: mockBuildingsList },
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "manager", "loggedIn": "true"}}',
  },
};
