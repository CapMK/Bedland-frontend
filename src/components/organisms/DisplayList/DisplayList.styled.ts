import styled from 'styled-components';

export const DisplayContainer = styled.div`
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit, minmax(20rem, 1fr));
`;

export default DisplayContainer;
