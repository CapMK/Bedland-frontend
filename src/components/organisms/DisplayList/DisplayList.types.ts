import { Building, ResidentsFlats } from 'types/shared.types';

export type DisplayListType = Building[] | ResidentsFlats[];

export type DisplayListProps = { displayList: DisplayListType };
