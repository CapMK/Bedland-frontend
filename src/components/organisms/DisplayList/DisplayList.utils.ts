const everythingAfterZipCodeRegex = /[0-9]{2}-[0-9]{3}.*$/;

const getCity = (address: string): string => {
  const cityWithZipCode = address.match(everythingAfterZipCodeRegex);
  if (cityWithZipCode) return cityWithZipCode[0].slice(7);
  return address.substring(address.lastIndexOf(',') + 2, address.length);
};

export default getCity;
