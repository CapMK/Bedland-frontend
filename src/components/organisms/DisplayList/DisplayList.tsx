import { useAppSelector } from 'store/hooks';
import Tile from 'components/molecules/Tile/Tile';
import AddAnotherTile from 'components/molecules/Tile/AddAnotherTile';
import { DisplayListProps } from './DisplayList.types';
import * as Styled from './DisplayList.styled';
import getCity from './DisplayList.utils';

const DisplayList = ({ displayList }: DisplayListProps) => {
  const userRole = useAppSelector((state) => state.user.role);
  const isManager = userRole === 'manager';
  const keyAttribute = isManager ? 'building' : 'flat';

  return (
    <Styled.DisplayContainer role='list' aria-labelledby='displaylist'>
      {displayList &&
        displayList.map((listItem) => (
          <Tile
            key={`${keyAttribute}-${'id' in listItem ? listItem.id : listItem.flatId}`}
            title={'name' in listItem ? listItem.name : `Flat #${listItem.flatNumber}`}
            subtitle={
              'address' in listItem ? `${getCity(listItem.address)}, Poland` : listItem.flatAddress
            }
            id={'id' in listItem ? listItem.id : listItem.flatId}
          />
        ))}
      {displayList.length < 5 && <AddAnotherTile />}
    </Styled.DisplayContainer>
  );
};
export default DisplayList;
