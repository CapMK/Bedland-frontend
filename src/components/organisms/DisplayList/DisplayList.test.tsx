import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import { loggedInManagerUser, loggedInResidentUser } from 'utils/mockUser';
import mockBuildingsList from 'utils/mock/mockBuildingsList';
import mockResidentFlats from 'utils/mock/mockResidentsFlats';
import DisplayList from './DisplayList';
import getCity from './DisplayList.utils';

describe('DisplayList', () => {
  afterEach(cleanup);
  it('buildings list renders correctly', () => {
    const buildingsList = renderWithProviders(<DisplayList displayList={mockBuildingsList} />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    expect(buildingsList).toMatchSnapshot();
  });

  it('flats list renders correctly', () => {
    const flatsList = renderWithProviders(<DisplayList displayList={mockResidentFlats} />, {
      preloadedState: {
        user: loggedInResidentUser,
      },
    });
    expect(flatsList).toMatchSnapshot();
  });

  it('getCity gets city from address with zip code', () => {
    const city = getCity('ulica Kruczkowskiego 001, 10-029 Opole');
    expect(city).toBe('Opole');
  });

  it('getCity gets city from address without zip code', () => {
    const city = getCity('ulica Kruczkowskiego 001, Opole');
    expect(city).toBe('Opole');
  });
});
