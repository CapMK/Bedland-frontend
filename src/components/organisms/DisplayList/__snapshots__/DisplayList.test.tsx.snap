// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayList > buildings list renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c4 {
  margin-top: 1rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c5:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c1 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c3 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c0 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

<body>
    <div>
      <div
        aria-labelledby="displaylist"
        class="c0"
        role="list"
      >
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="building"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Business Garden Wrocław
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Wrocław, Poland
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/1"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="building"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Business Garden Poznań
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Poznań, Poland
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/2"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="building"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Business Park
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Kraków, Poland
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/3"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="building"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Business Garden
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Opole, Poland
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/4"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="building"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Business Garden Warszawa
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              Warszawa, Poland
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/5"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c4 {
  margin-top: 1rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c5:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c1 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c3 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c0 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

<div>
    <div
      aria-labelledby="displaylist"
      class="c0"
      role="list"
    >
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="building"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Business Garden Wrocław
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Wrocław, Poland
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/1"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="building"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Business Garden Poznań
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Poznań, Poland
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/2"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="building"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Business Park
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Kraków, Poland
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/3"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="building"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Business Garden
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Opole, Poland
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/4"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="building"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Business Garden Warszawa
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Warszawa, Poland
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/5"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`DisplayList > flats list renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c4 {
  margin-top: 1rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c5:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c1 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c3 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c0 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

<body>
    <div>
      <div
        aria-labelledby="displaylist"
        class="c0"
        role="list"
      >
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="flat"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Flat #1
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              ul. Legnicka 48H, 54-202 Wrocław
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/1"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="flat"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Flat #2
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              ul. Legnicka 48H, 54-202 Wrocław
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/2"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="flat"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Flat #3
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              ul. Legnicka 48H, 54-202 Wrocław
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/3"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="flat"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Flat #4
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              ul. Legnicka 48H, 54-202 Wrocław
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/4"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="c1"
          role="listitem"
        >
          <img
            alt="flat"
            class="c2"
            src="/src/assets/img/BuildingImg.png"
          />
          <div
            class="c3"
          >
            <h3
              style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
            >
              Flat #1
            </h3>
            <p
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              ul. Kolorowa 6, 60-198 Poznań
            </p>
            <div
              class="c4"
            >
              <a
                class=""
                href="/buildings/5"
              >
                <button
                  class="c5"
                >
                  Open details
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c4 {
  margin-top: 1rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c5:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c1 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c3 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c0 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

<div>
    <div
      aria-labelledby="displaylist"
      class="c0"
      role="list"
    >
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="flat"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Flat #1
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            ul. Legnicka 48H, 54-202 Wrocław
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/1"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="flat"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Flat #2
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            ul. Legnicka 48H, 54-202 Wrocław
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/2"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="flat"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Flat #3
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            ul. Legnicka 48H, 54-202 Wrocław
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/3"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="flat"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Flat #4
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            ul. Legnicka 48H, 54-202 Wrocław
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/4"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="c1"
        role="listitem"
      >
        <img
          alt="flat"
          class="c2"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c3"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            Flat #1
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            ul. Kolorowa 6, 60-198 Poznań
          </p>
          <div
            class="c4"
          >
            <a
              class=""
              href="/buildings/5"
            >
              <button
                class="c5"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
